lib/include/prjxray/xilinx/spartan6/bitstream_reader.h

```diff
@@ -0,0 +1,94 @@
+#ifndef PRJXRAY_LIB_XILINX_XC7SERIES_BITSTREAM_READER_H
+#define PRJXRAY_LIB_XILINX_XC7SERIES_BITSTREAM_READER_H
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+#include <absl/types/optional.h>
+#include <absl/types/span.h>
+
+#include <prjxray/big_endian_span.h>
+#include <prjxray/xilinx/spartan6/configuration_packet.h>
+
+namespace prjxray {
+namespace xilinx {
+namespace xc7series {
```

Comment:
Is the idea to leave the name space as xc7series until we merge everything back together?

Reply:
Not really, I focused on getting bitread to correctly read a Spartan6 bitstream and produce a sensible bits file first. I will focus on refactoring once I am sure this works fine, .i.e get the sp6patch to write a valid bitstream with the data obtained from bitread (since the bitstream reader class is in a common library for both tools). Expect a PR with sp6patch soon.